### PR TITLE
feat: extract List to a separate package

### PR DESCRIPTION
**Pull request by nikazawila**
This moves `List` to a separate component for v4

### CODE REVIEW

File: packages/components/list/src/List.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }),
+        className,
+      )}
+      ref={ref}
+      {...otherProps}
+    >
+      {children}
+    </Tag>
+  );
+}
```
Comment (denkristoffer) [inline packages/components/list/src/List.tsx:43]: I think we should use our polymorphic components here, maybe just a `Box` that can be either `ul` or `ol`? That would also give us the shared props like `testId` and `className` 🙂

**Review by denkristoffer — CHANGES_REQUESTED** [review]
Thanks for tackling this 🙏 I think we should take advantage of our polymorphic building block components here, probably the `Box` in this case.

**Comment by nikazawila** [comment]
I'd appreciate some help on figuring out what I did wrong that the deployment fails...

File: packages/components/list/src/ListItem/ListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { List } from '../List';
+
+export interface ListItemProps {
+  // Additional className
+  className?: string;
+  // The children of the ListItem
+  children: React.ReactNode;
+  // Class used for testing purposes
+  testId?: string;
+}
```
Comment (denkristoffer) [inline packages/components/list/src/ListItem/ListItem.tsx:14]: I think we should also use `CommonProps` for this component.

File: packages/components/list/src/List.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+export type ListProps = PolymorphicComponentProps<
+  typeof DEFAULT_TAG,
+  ListInternalProps
+>;
+
+const _List: PolymorphicComponentWithRef<
+  ListInternalProps,
+  typeof DEFAULT_TAG
+> = ({ as: Tag = 'ul', className, children, ...otherProps }, ref) => {
```
Comment (denkristoffer) [inline packages/components/list/src/List.tsx:28]:
```suggestion
> = ({ as: Tag = DEFAULT_TAG, className, children, ...otherProps }, ref) => {
```

Comment (denkristoffer) [reply packages/components/list/src/List.tsx:28]: One last thing 🙂

**Review by denkristoffer — APPROVED** [review]

**Pull request merged**